sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -208,3 +208,161 @@ object ExtractPythonUDFFromJoinCondition extends Rule[LogicalPlan] with Predicat
       }
   }
 }
+
+sealed abstract class BuildSide
+
+case object BuildRight extends BuildSide
+
+case object BuildLeft extends BuildSide
+
+trait JoinSelectionHelper {
+
+  def getBroadcastBuildSide(
+      left: LogicalPlan,
+      right: LogicalPlan,
+      joinType: JoinType,
+      hint: JoinHint,
+      onlyLookingAtHint: Boolean,
```

Comment:
nit: `selectOnlyFromHint`? I couldn't tell what this meant just by the name `onlyLookingAtHint `.

Reply:
If we want to be super clear, maybe "pickJoinSideByHintOnly"? or simply "hintOnly".

Reply:
Ah, `hintOnly` looks nice to me.